react/features/invite/components/security-dialog/PasswordSection.js

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable react/no-multi-comp */
+// @flow
+import React, { useRef } from 'react';
+
+import PasswordForm from './PasswordForm';
+import { translate } from '../../../base/i18n';
+import { copyText } from '../add-people-dialog';
+
+type Props = {
+
+    /**
+     * Whether or not the current user can modify the current password.
+     */
+    canEditPassword: boolean,
+
+    /**
+     * The JitsiConference for which to display a lock state and change the
+     * password.
+     */
+    conference: Object,
+
+    /**
+     * The value for how the conference is locked (or undefined if not locked)
+     * as defined by room-lock constants.
+     */
+    locked: string,
+
+    /**
+     * The current known password for the JitsiConference.
+     */
+    password: string,
+
+    /**
+     * Whether or not to show the password in editing mode.
+     */
+    passwordEditEnabled: boolean,
+
+    /**
+     * The number of digits to be used in the password.
+     */
+    passwordNumberOfDigits: ?number,
+
+    /**
+     * Action that sets the conference password.
+     */
+    setPassword: Function,
+
+    /**
+     * Method that sets whether the password editing is enabled or not.
+     */
+    setPasswordEditEnabled: Function,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+};
+
+/**
+ * Component that handles the password manipulation from the invite dialog.
+ *
+ * @returns {React$Element<any>}
+ */
+function PasswordSection({
+    canEditPassword,
+    conference,
+    locked,
+    password,
+    passwordEditEnabled,
+    passwordNumberOfDigits,
+    setPassword,
+    setPasswordEditEnabled,
+    t }: Props) {
+    const formRef = useRef(null);
+
+    /**
+     * Callback invoked to set a password on the current JitsiConference.
+     *
+     * @param {string} enteredPassword - The new password to be used to lock the
+     * current JitsiConference.
+     * @private
+     * @returns {void}
+     */
+    function onPasswordSubmit(enteredPassword) {
+        setPassword(
+            conference,
+            conference.lock,
+            enteredPassword
+        );
+    }
+
+    /**
+     * Toggles whether or not the password should currently be shown as being
+     * edited locally.
+     *
+     * @private
+     * @returns {void}
+     */
+    function onTogglePasswordEditState() {
+        setPasswordEditEnabled(!passwordEditEnabled);
+    }
+
+    /**
+     * Method to remotely submit the password from outside of the password form.
+     *
+     * @private
+     * @returns {void}
+     */
+    function onPasswordSave() {
+        // $FlowFixMe
+        formRef.current.querySelector('form').requestSubmit();
+    }
+
+    /**
+     * Callback invoked to unlock the current JitsiConference.
+     *
+     * @private
```

Comment:
I generally don't like to use these `@private` flags (I think I actually never use them), because I deny the existence of private functions in JS (it's not TS, unfortunately). But if you like to use them, it's fine for me, just be consistent - e.g. this one is private, but the next two are not, whereas I think they are of the same theoretical visibility.

Reply:
I actually agree with your opinion, sometimes copy-pasta goes wrong though...